### ISSUE TITLE
New embedded KubernetesClient broken for KubeOps v7

### ISSUE THREAD

**Issue by Crespalvaro** [buehler/dotnet-operator-sdk#513]
**Describe the bug**
The mapping between the different Kubernetes Objects is not working as expected. Previously the generic methods of this client would allow to operate with any object from the cluster, regardless of the kind, group and version, as long as this properties are defined in the Type, either Custom Resources or k8s.models, and the Type implements IKuebernetesObject<V1ObjectMeta>.

However, at least since KubeOps v7 there are issues with the usage of the client, for instance, not being able to find an object in the cluster

**To Reproduce**
Find below a code section to try and reproduce
```
//Normally attempt to retrieve an existing secret
var res = await _client.Get<V1Secret>("foo", "bar"); //res == null

//Get maps this request to ReadAsync/ReadNamespacedAsync based on namespace == null
//Since its a namespace3d object, and namespace prop is provided, the mapping should resolve in the next call to =>
var res2 = await ApiClient.CustomObjects.GetNamespacedCustomObjectWithHttpMessagesAsync(V1Secret.KubeGroup, V1Secret.KubeApiVersion, "bar", V1Secret.KubePluralName, "foo");

res = KubernetesJson.Deserialize<V1Secret>(res2.Body.ToString()); 
//res contains the actual secrect successfully retrieved from the cluster
```

**Additional context**
I want to take the change to translate a doubt. Wich is the point on setting private the old k8s base ApiClient, now renamed to [_client](https://github.com/buehler/dotnet-operator-sdk/blob/164ec24e1fb637579c20787fde1a4e0263511ac2/src/KubeOps.KubernetesClient/KubernetesClient.cs#L16). The ApiClient is a wide source of opportunities to access direct methods as well as avoid KubeOps to keep/wrap all the k8s requests up to date. I now it is quite easy to instantiate your own instance of it, thus Im doing so. However, it is quite awfull

**Comment by Crespalvaro** [comment]
Clossing this issue since fix is provided at version 7.0.2

**Comment by buehler** [comment]
Hey @Crespalvaro 
Yes, the issue was brought up (and fixed) by georgevella in #511.
